clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1145,6 +1150,7 @@ public void onFailure(RuntimeException e) {
                 if (commitException instanceof FencedInstanceIdException) {
                     asyncCommitFenced.set(true);
                 }
+                inFlightAsyncCommits.decrementAndGet();
```

Comment:
I'm not sure if we need to wrap the `decrementAndGet` call in a `try`/`finally` block here, too?

Reply:
I think it is not needed here. There is nothing that could fail with an exception.